### PR TITLE
use ANSI/C89 function signatures

### PR DESCRIPTION
**Pull request by alanpost**
This PR is part of the de-K&R series, here adding ANSI (C89) function signatures.

### CODE REVIEW

File: coe.c [hunk 1]
```diff
@@ -1,8 +1,8 @@
+#include <unistd.h>
```
Comment (josuah) [inline coe.c:1]: - https://pubs.opengroup.org/onlinepubs/9699919799/functions/fcntl.html
- http://man.openbsd.org/fcntl
- https://www.freebsd.org/cgi/man.cgi?query=fcntl
- https://netbsd.gw.com/cgi-bin/man-cgi?fcntl++NetBSD-current

None of these state to #include <unistd.h>, but glibc man pages does. Cannot harm to add an extra header...

File: qmail-qmqpc.c [hunk 2]
```diff
@@ -135,7 +135,8 @@ char *server;
 
 stralloc servers = {0};
 
-main()
+void
```
Comment (josuah) [inline qmail-qmqpc.c:138]: I personally use this style, but this has the side effect of shifting the line numbers (does it actually matter for patch?) and is not consistent with the rest of the source, with type and function name on the same line.

File: quote.h [hunk 3]
```diff
@@ -1,8 +1,10 @@
 #ifndef QUOTE_H
 #define QUOTE_H
 
-extern int quote_need();
-extern int quote();
-extern int quote2();
+#include "stralloc.h"
```
Comment (josuah) [inline quote.h:4]: As a good side effect, we notice more dependencies between source doe files.  Maybe will help to generate the Makefile later...

File: qmail-qmqpc.c (hunk 2 above)
Comment (DerDakon) [reply qmail-qmqpc.c:138]: The line number change will not harm, patch will just output a message about offset.

**Comment by DerDakon** [comment]
Meanwhile we agreed that we will only touch those function signatures that we change for one or the other reason anyway to reduce conflicts with external patches. Since this has it's own amount of collisions I'll close it for now.

**Pull request closed without merging**